backend/services/aws_service.py

```
@@ -0,0 +1,139 @@
+            region_name=region
+        )
+        
+        log.info(f"AWS service initialized for region: {region}")
+    
+    def list_s3_files(self, bucket: str) -> List[str]:
+        """List objects in S3 bucket"""
+        try:
+            response = self.s3_client.list_objects_v2(Bucket=bucket)
+            files = [obj["Key"] for obj in response.get("Contents", [])]
+            log.info(f"Listed {len(files)} files from bucket {bucket}")
+            return files
```

Ruff (0.13.1)

Consider moving this statement to an `else` block

(TRY300)